src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -304,10 +304,9 @@ def __reconstruct_malicious(
                 all_shares[(party_rank + 1) % (nparties)]
             )
 
-            if not value:
+            if value is None:
                 value = share_sum
-
-            elif share_sum != value:
+            elif (share_sum != value).any():
```

Comment:
Q: Why you have .any() ?

Reply:
If any value in share_sum and value are not equal to each other it should evaluate to true.